src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1256,6 +1268,40 @@ public function getExpressionLanguageProviders()
     }
 
     /**
+     * Returns a ChildDefinition that will be used for autoconfiguring the interface/class.
+     *
+     * @param string $interface The class or interface to match
+     * @return ChildDefinition
+     */
+    public function registerForAutoconfiguration($interface)
+    {
+        if (!isset($this->automaticInstanceofDefinitions[$interface])) {
+            $this->automaticInstanceofDefinitions[$interface] = array();
+        }
+
+        $childDefinition = new ChildDefinition('');
+        $this->automaticInstanceofDefinitions[$interface][] = $childDefinition;
```

Comment:
Definitely appending (which means, merging). 

Why?

1) If 2 DI extensions both autoconfigure the same instance (which probably shouldn't happen), we should probably **throw an exception** or **merge** them together. Replacing seems completely wrong, since execution order isn't guaranteed. Of course, merging is a bit weird too, since execution order isn't guaranteed. But if they configure different things, merging is great. If they configure the same config... well, they shouldn't do that :).

2) If the end-user wants to auto configure an instanceof globally, why not let them? e.g. suppose they want to add some extra config to all VoterInterface? In this case, throwing an exception does *not* make sense. So, **replacing** or **merging** makes sense. And I think merging is more user-friendly: they're able to inherit all the original auto-config, then override whatever they want. They will get duplicate tags... but since new tags will be added first, they will override the original tags.

So, appending/merging makes sense to me. This actually affects 2 places in the code - calling `registerForAutoconfiguration()` 2 times on the same builder and merging 2 builders... but I think they should probably act the same.